controller/model/posture_check_model_process_test.go

```diff
@@ -0,0 +1,195 @@
+
+		result := processCheck.Evaluate(postureData)
+		req := require.New(t)
+
+		req.True(result)
+	})
+
+	t.Run("returns true for valid id, running, hash, and fingerprint with mismatched signer case", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		postureData.Processes[0].SignerFingerprint = strings.ToUpper(postureData.Processes[0].SignerFingerprint)
+
+		result := processCheck.Evaluate(postureData)
+		req := require.New(t)
+
+		req.True(result)
+	})
+
+	t.Run("returns true for valid id and running but check has null hashes and no signer", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		processCheck.Hashes = nil
+		processCheck.Fingerprint = ""
+
+		result := processCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	t.Run("returns true for valid id and running but check has empty hashes and no signer", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		processCheck.Hashes = []string{}
+		processCheck.Fingerprint = ""
+
+		result := processCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	t.Run("returns true for valid id, running, signer but check has no hash", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		processCheck.Hashes = nil
+
+		result := processCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	t.Run("returns true for valid id, running, hashes but check has no signer", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		processCheck.Fingerprint = ""
+
+		result := processCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	t.Run("returns false if ids do not match", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		processCheck.PostureCheckId = "does not match"
+
+		result := processCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.False(result)
+	})
+
+	t.Run("returns false if signers do not match", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		postureData.Processes[0].SignerFingerprint = "does not match"
+
+		result := processCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.False(result)
+	})
+
+	t.Run("returns false if hashes do not match", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		postureData.Processes[0].BinaryHash = "does not match"
+
+		result := processCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.False(result)
+	})
+
+	t.Run("returns false not running, invalid hash, invalid signer", func(t *testing.T) {
+		processCheck, postureData := newMatchingProcessCheckAndData()
+		postureData.Processes[0].IsRunning = false
+		postureData.Processes[0].BinaryHash = "does not match"
+		postureData.Processes[0].BinaryHash = "does not match"
```

Comment:
duplicated line that doesn't have any effect :)

Reply:
Fixed.